wolfcrypt/src/tfm.c

```diff
@@ -1868,9 +1868,23 @@ void fp_montgomery_reduce(fp_int *a, fp_int *m, fp_digit mp)
 
 void fp_read_unsigned_bin(fp_int *a, const unsigned char *b, int c)
 {
+    /* must leave room for extra digit because of fp_mul_2d below */
```

Comment:
I confirmed multiple ways the overflow occurs when at the FP_SIZE max without the -1, but you make a good point that fp_mul_2d shouldn't allow it. I'll investigate further. Also I'm doing a math lib update so I might find some other things.